packages/firestore/src/local/local_store.ts

```diff
@@ -1057,152 +1048,137 @@ export function newLocalStore(
   return new LocalStoreImpl(persistence, queryEngine, initialUser);
 }
 
-/**
- * An interface on top of LocalStore that provides additional functionality
- * for MultiTabSyncEngine.
- */
-export interface MultiTabLocalStore extends LocalStore {
-  /** Returns the local view of the documents affected by a mutation batch. */
-  lookupMutationDocuments(batchId: BatchId): Promise<MaybeDocumentMap | null>;
-
-  removeCachedMutationBatchMetadata(batchId: BatchId): void;
-
-  setNetworkEnabled(networkEnabled: boolean): void;
-
-  getActiveClients(): Promise<ClientId[]>;
-
-  getTarget(targetId: TargetId): Promise<Target | null>;
-
-  /**
-   * Returns the set of documents that have been updated since the last call.
-   * If this is the first call, returns the set of changes since client
-   * initialization. Further invocations will return document changes since
-   * the point of rejection.
-   */
-  getNewDocumentChanges(): Promise<MaybeDocumentMap>;
-
-  /**
-   * Reads the newest document change from persistence and forwards the internal
-   * synchronization marker so that calls to `getNewDocumentChanges()`
-   * only return changes that happened after client initialization.
-   */
-  synchronizeLastDocumentChangeReadTime(): Promise<void>;
+/** Returns the local view of the documents affected by a mutation batch. */
+// PORTING NOTE: Multi-Tab only.
+export function lookupMutationDocuments(
+  localStore: LocalStore,
+  batchId: BatchId
+): Promise<MaybeDocumentMap | null> {
+  const localStoreImpl = debugCast(localStore, LocalStoreImpl);
+  const mutationQueueImpl = debugCast(
+    localStoreImpl.mutationQueue,
+    IndexedDbMutationQueue // We only support IndexedDb in multi-tab mode.
+  );
```

Comment:
This setup is being done in every reworked method, would it be possible to refactor it out into a helper function returning two values?

Reply:
It's not that pretty since it needs a cast in the method call, but I changed it:

```
const [localStoreImpl, mutationQueueImpl] = 
    debugCast(localStore, LocalStoreImpl,  (localStoreImpl as LocalStoreImpl).mutationQueue, IndexedDbMutationQueue);
```

Reply:
Hm, I undid this since this pattern actually uses one more line for each invocation.

Reply:
Hmm, what I had in mind was just
```typescript
const [localStoreImpl, mutationQueueImpl] = foo(localStore);
```
with the helper function taking care of everything else.

Reply:
I can add a `multiTabLocalStore` function that returns LocalStoreImpl, IndexedDbRemoteDocumentCache, IndexedDbMutationQueue and IndexedDbTargetCache. Does that sound good to you?